check.py

```diff
@@ -321,6 +321,25 @@ def do_asm2wasm_test():
     with open(out) as f:
       fail_if_not_identical(f.read(), actual)
 
+print '\n[ checking wasm-reduce ]\n'
+
+# check if we have `timeout`, which the reducer depends on
+has_shell_timeout = (os.system('timeout 1s pwd') == 0)
```

Comment:
It's more specific than that, it also needs a full-featured `timeout` utility, which e.g. the ARM linux variant on the bots lack. Refactoring this check into a shared location.